src/public_parameters/disk_cache.rs

```diff
@@ -6,13 +6,18 @@ use abomonation::{encode, Abomonation};
 use camino::{Utf8Path, Utf8PathBuf};
 use nova::traits::Group;
 
+use crate::config::LURK_CONFIG;
 use crate::coprocessor::Coprocessor;
 use crate::proof::nova::{CurveCycleEquipped, PublicParams, G1, G2};
 use crate::proof::MultiFrameTrait;
 use crate::public_parameters::error::Error;
 
 use super::instance::Instance;
 
+pub(crate) fn public_params_dir() -> Utf8PathBuf {
```

Comment:
I think this function can just return `&Utf8PathBuf` to avoid that clone. Does any consumer of this data needs ownership?